packages/backend/src/functions/httpTrigger.ts

```diff
@@ -187,9 +194,63 @@ async function postProvenance(request: HttpRequest, context: InvocationContext):
         const provRecord: ProvenanceRecord = { record, attachments };
         const data = new TextEncoder().encode(JSON.stringify(provRecord));
         const recordID = await upload(containerClient, deviceKey, data, "prov", "application/json", timestamp);
-        return { jsonBody: { record: recordID, attachments } };
+      return {
+        headers: { "Access-Control-Allow-Origin" : "*"},
+        jsonBody: { record: recordID, attachments } };
     }
 }
+// blobNames typically look like: 'gosqas/63f4b781c0688d83d40908ff368fefa6a2fa4cd470216fd83b3d7d4c642578c0/prov/1a771caa4b15a45ae97b13d7a336e1e9c9ec1c91c70f1dc8f7749440c0af8114'
+// where the id is that last part (before the last slash)
+function findDeviceIdFromName(blobName : string) : string {
+    return blobName.split("/",4)[3];
+}
+
+async function getStatistics(request: HttpRequest, context: InvocationContext): Promise<HttpResponseInit> {
+    let i = 1;
+    // some options for filtering list
+    const listOptions = {
+        includeMetadata: true,
+        includeSnapshots: false,
+        includeTags: false,
+        includeVersions: false,
+        prefix: ''
+    };
+
+    // Build up a JSON return value
+    // NOTE: We seem to have to read the properties of the blob to get the
+    // metadata.  There is a field called "metadata" on the blob itself
+    // which does not contain our metadata. I don't know if this is terribly
+    // expensive, or if we could improve it. I insist we should not worry about
+    // performance until we measure it to be a problem, but this is an "orang flag"--
+    // some caution around this issue is warranted.
+    var records = [];
+    for await (const blob of containerClient.listBlobsFlat()) {
+        const blobClient = containerClient.getBlockBlobClient(blob.name);
+        const props = await blobClient.getProperties();
+        const metadata = props.metadata;
+        // now we want to build up an object that we can return as statistics
+        // that inlcudes the id and the timestamp, though really the timestamp
+        // is enough. We would like to distinguish the additon of a device
+        // from the addition of new provenance, I supoose.
+        const id = findDeviceIdFromName(blob.name);
+        // We could do some sorting in this function, but that is more or less
+        // easily done by whomever is using this. So I think it better to just
+        // return the data in  a fairly raw form, as an array of {timestamp, id} tuples.
+        // Eventually, this function may have to only look back X days or X hours,
+        // but until it gets unwieldy we can return everything.
+        // I think the proper way to test this is to build a test program that
+        // puts 1000s of objects into the database and see where performance becomes a problem.
+        records.push({ gdttimestamp: metadata.gdttimestamp, gdtid: id});

```

Comment:
we probably don't need `gdt` prefixes here. And I think we should use the full name `deviceID` not just `id` for this record

Reply:
This has been changed to "timestamp" and "deviceID" as requested.